app/controllers/reliefs/AvoidanceSchemeBeingUsedController.scala

```diff
@@ -16,46 +16,47 @@
 
 package controllers.reliefs
 
-import config.FrontendDelegationConnector
 import connectors.{BackLinkCacheConnector, DataCacheConnector}
-import controllers.{AtedBaseController, BackLinkController}
-import controllers.auth.{AtedFrontendAuthHelpers, AtedRegime, ClientHelper}
+import controllers.BackLinkController
+import controllers.auth.{AuthAction, ClientHelper}
 import forms.ReliefForms._
 import models.IsTaxAvoidance
-import services.ReliefsService
-import uk.gov.hmrc.play.frontend.auth.DelegationAwareActions
-import utils.{AtedUtils, PeriodUtils}
-import play.api.i18n.Messages.Implicits._
 import play.api.Play.current
+import play.api.i18n.Messages.Implicits._
+import play.api.mvc.{Action, AnyContent}
+import services.{DelegationService, ReliefsService}
+import utils.{AtedUtils, PeriodUtils}
 
 import scala.collection.mutable.ArrayBuffer
 import scala.concurrent.Future
 
 trait AvoidanceSchemeBeingUsedController extends BackLinkController
-  with AtedFrontendAuthHelpers with ReliefHelpers with DelegationAwareActions with ClientHelper {
+   with ReliefHelpers with ClientHelper with AuthAction {
 
-  def view(periodKey: Int) = AuthAction(AtedRegime) {
-    implicit atedContext =>
+  def view(periodKey: Int): Action[AnyContent] = Action.async { implicit request =>
+    authorisedAction { implicit authContext =>
       ensureClientContext {
         validatePeriodKey(periodKey) {
           for {
             backLink <- currentBackLink
-            retrievedData <- reliefsService.retrieveDraftReliefs(atedContext.user.atedReferenceNumber, periodKey)
+            retrievedData <- reliefsService.retrieveDraftReliefs(authContext.atedReferenceNumber.get, periodKey)
```

Comment:
RED - None.get risk.

Reply:
Removed and refactored to use String as opposed to Option String